Assets/Scripts/ExerciseManager.cs

```diff
@@ -16,27 +19,44 @@ public class ExerciseManager : Singleton<ExerciseManager> {
 #if WINDOWS_UWP
             curExercisePos = 0;
 #endif
+            SetupCodeBlockDictionary();
             SetUpExerciseList();
-            curExercise = exerciseList[curExercisePos];
+            fileEntries = Directory.GetFiles(targetDirectory).Where(s => s.EndsWith(".json")).ToArray();
+            if(curExercisePos < fileEntries.Length) {
+                SetUpCurExercise(curExercisePos);
+            } else {
+                InitiateFreePlay();
+            }
             ToggleCurrentExercise(true);
             LoggingManager.instance.AddLogColumn(curExcersieCol, curExercisePos.ToString());
             LoggingManager.instance.AddLogColumn(exerciseSubmissionResultCol, "");
         }
 
-        private void SetUpExerciseList() {
-            exerciseList = new List<Exercise>();
-            foreach (Exercise ex in GetComponentsInChildren<Exercise>(true)) {
-                exerciseList.Add(ex);
-                ex.gameObject.SetActive(false);
-            }
+        private void SetUpCurExercise(int exerciseNum) {
+            string json = File.ReadAllText(fileEntries[exerciseNum]);
+            GameObject exercise = InstantiateExercise(json);
+            //Add scaffoldDialogue
+            exercise.AddComponent<ExerciseScaffolding>().SetScaffoldDialogue(
+                curExercise.GetExerciseInternals().GetscaffoldDialogue());
+            //TODO: add ExerciseInformationSeekingActions
         }
 
         public Exercise GetCurExercise() {
             return curExercise;
         }
 
+        public GameObject InstantiateExercise(string jsonString) {
+            GameObject exGO = Instantiate(
+                Resources.Load<GameObject>(ResourcePathConstants.ExercisePrefab), transform.parent) as GameObject;
+            exGO.transform.SnapToParent(transform);
+            exGO.GetComponent<Exercise>().SetupExercise(jsonString);
+            curExercise = exGO.GetComponent<Exercise>();
+            exGO.GetComponent<Exercise>().SetUpKuriInExercise();
+            return exGO;
+        }
+
         public void AlertCodeFinished() {
-            if (curExercise != null) { // This if is to guard against initializing interpreter
+            if (curExercise != null && curExercise.GetType() != typeof(FreePlayExercise)) { // This if is to guard against initializing interpreter
```

Comment:
ah I see what you did here; I guess that works still as opposed to just always saying freeplay is wrong